recipes/intel-media-driver/all/conanfile.py

```diff
@@ -0,0 +1,95 @@
+from conan import ConanFile
+from conan.tools.files import copy, get, collect_libs, rm, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps, cmake_layout
+from conan.tools.env import Environment, VirtualBuildEnv
+from conan.tools.gnu import PkgConfigDeps
+from conan.errors import ConanInvalidConfiguration
+import os
+import textwrap
+import shutil
+
+required_conan_version = ">=1.53"
+
+
+class IntelMediaDriverConan(ConanFile):
+    name = "intel-media-driver"
+    description = "VAAPI drivers for Intel graphics processors"
+    topics = ("intel", "vaapi")
+    homepage = "https://github.com/intel/media-driver"
+    url = "https://gitlab.com/missionrobotics/conan-packages/mr-conan-index"
+    license = "MIT"
+    settings = "os", "arch", "compiler", "build_type"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True,
+    }
+
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Windows is not supported")
+        if self.info.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, 11)
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+    
+    def layout(self):
+        cmake_layout(self)
+
+    def requirements(self):
+        self.requires( "libva/2.16.0" )
+        self.requires( "gmmlib/22.3.3" )
+
+    def build_requirements(self):
+        self.tool_requires("pkgconf/1.9.3")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
+
+        pc = PkgConfigDeps(self)
+        pc.generate()
+
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+        # Override Avahi's problematic check for the pkg-config executable.
+        env = Environment()
+        env.define("have_pkg_config", "yes")
+        env.vars(self).save_script("conanbuild_pkg_config")
+
+    def package(self):
+        copy(self, "LICENSE.md", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "intel-media-driver")
+        self.cpp_info.set_property("cmake_target_name", "intel-media-driver")
+        self.cpp_info.set_property("pkg_config_name", "intel-media-driver")
+
+        self.cpp_info.components["intel-media-driver"].libs = collect_libs(self)
```

Comment:
You should avoid using `collect_libs` if possible. Since this is a runtime-loadable plugin, I'm not sure that you even need to provide a library in this way.

You'll want to make the VA-API libraries accessible at runtime:

```
self.runenv_info.prepend_path("LIBVA_DRIVERS_PATH", os.path.join(self.package_folder, "lib", "dri"))
```

Reply:
Yeah, I think it might be worth setting here and letting consumer's figure out whether or not to include it as a dependency as appropriate for the specific system.

Reply:
One thought just now, how would you expect to handle the case where there are two VAAPI driver packages that both do this?

As an example, in our app we pull in both `intel-media-driver` and `nvidia-vaapi-driver` and the correct one can be used by the application based on the hardware it is running on. There can only be one `LIBVA_DRIVERS_PATH`, so it seems like the value couldn't be the individual package folder itself in that case?

Reply:
Seriously? That seems like a ridiculous limitation. Do you know if there's a good reason for that?

Reply:
Hmm, I found this: https://github.com/canonical/snapcraft/pull/3591/files which seems to indicate someone at least tried to append a path to that variable.

Reply:
Aha! It should work with multiple paths: https://github.com/intel/libva/issues/639#issuecomment-1385276574

Reply:
https://github.com/intel/libva/blob/907b2b5405ca1091b4360bf35060e143bd704b62/va/va.c#L393
Great thanks, just validated it myself in the source. Good find!